src/hook.c

```diff
@@ -757,3 +761,46 @@ MH_STATUS WINAPI MH_ApplyQueued(VOID)
         LeaveSpinLock();
     }
 }
+
+//-------------------------------------------------------------------------
+MH_STATUS WINAPI MH_CreateHookApi(LPCSTR pszTarget, LPVOID pDetour, LPVOID *ppOriginal)
+{
+    HMODULE hModule;
+    LPVOID  pTarget;
+    SIZE_T  copySize;
+    LPCSTR  psz = pszTarget;
+    LPCSTR  pszProcName = NULL;
+    CHAR    szModuleName[MAX_PATH];
+
+    if (pszTarget == NULL || pDetour == NULL)
+        return MH_ERROR_INVALID_ARGUMENTS;
+
+    // search the last '.' (replacement for strrchr())
+    while (*psz)
+        if (*psz++ == '.')
+            pszProcName = psz;
```

Comment:
Yes, you're right. My bad. Sorry for saying nonsense about the code.

> you forgot that:
> &MessageBoxW != GetProcAddress(GetModuleHandleA("user32.dll"), "MessageBoxW")

Many times it is. Specifically for `MessageBoxW`: http://i.snag.gy/y23K1.jpg
Or do you mean something else?

> Privately, I often hook the functions of ntdll.dll (or "undocumented" functions) so find this helper useful.

Great, that means that the function is useful. You can disregard my first comment.

P.S. is there a reason not to use two arguments, one for the DLL and one for the function name? i.e.:

    MH_STATUS WINAPI MH_CreateHookApi(LPCWSTR pszModule, LPCSTR pszProcName, LPVOID pDetour, LPVOID *ppOriginal);

IMO that will make the call clearer, more universal (Unicode file names), more performant (no search/copy needed), and will allow the compiler to reuse strings (i.e. it can reuse "user32.dll" which is used multiple times, but not when it's used in "user32.dll.xxx" with different xxx).